File: backend/app/schemas/weather.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime

class WeatherResponse(BaseModel):
    id: str
    city_id: str
    
    # Raw metrics
    temperature: float
    apparent_temperature: float
    rainfall: float
    precipitation_probability: int
    humidity: int
    wind_speed: float
    wind_direction: float
    weather_condition: str
    cloud_cover: int
    
    # Frontend derived values mapped from metrics
    label: str
    emoji: str
    alertText: str
    alertLevel: str
    
    last_updated: datetime

    class Config:
        orm_mode = True
        from_attributes = True

File: types/weather.ts
export interface Weather {
  id: string;
  city_id: string;
  temperature: number;
  apparent_temperature: number;
  rainfall: number;
  precipitation_probability: number;
  humidity: number;
  wind_speed: number;
  wind_direction: number;
  weather_condition: string;
  cloud_cover: number;
  label: string;
  emoji: string;
  alertText: string;
  alertLevel: string;
  last_updated: string;
}
